cohttp-eio/src/client.ml

```diff
@@ -25,12 +25,20 @@ type 'a body_allowed_call =
   response
 
 (* Request line https://datatracker.ietf.org/doc/html/rfc7230#section-3.1.1 *)
-let write_request writer (meth, version, headers, resource_path, body) =
-  Buf_write.string writer (Http.Method.to_string meth);
+let write_request writer request body =
+  let headers =
+    Body.add_content_length
+      (fun () -> Http.Request.requires_content_length request)
```

Comment:
Why you need to delay this and not simply pass the boolean?

Reply:
Indeed. This is just a `bool` now.